src/Mod/Draft/draftguitools/gui_hyperlink.py

```diff
@@ -0,0 +1,116 @@
+# ***************************************************************************
+
+"""Provides GUI tools to open hyperlinks to internal/external documents."""
+
+## @package gui_hyperlink
+# \ingroup draftguitools
+# \brief Provides GUI tools to open hyperlinks to internal/external documents
+
+## \addtogroup draftguitools
+# @{
+
+import FreeCAD
+import os, sys, subprocess
+import re
+import draftutils.utils as utils
+from draftutils.messages import _msg
+
+if FreeCAD.GuiUp:
+    import FreeCADGui
+    import Draft_rc
+def QT_TRANSLATE_NOOP(ctx,txt):
+    return txt
+translate = FreeCAD.Qt.translate
+
+from PySide import QtWidgets
+
+__title__ = "FreeCAD Draft Workbench GUI Tools - Hyperlinks tools"
+__author__ = ("")
+__url__ = "https://www.freecad.org"
+
+
+class Draft_Hyperlink:
+    """The Draft_Hyperlink FreeCAD command definition."""
+
+    def GetResources(self):
+        d = {'Pixmap': '',
+             'Accel': "",
+             'MenuText': QT_TRANSLATE_NOOP("Draft_Hyperlink", "Open hyperlinks"),
+             'ToolTip': QT_TRANSLATE_NOOP("Draft_Hyperlink", "Open linked documents")}
+        return d
+
+    def Activated(self):
+        self.find_hyperlinks()
+
+        if len(self.hyperlinks_list) > 1:
+            m = QtWidgets.QMessageBox()
+            m.setWindowTitle(translate("draft", "Opening multiple hyperlinks"))
+            m.setText(
+                translate(
+                    "draft",
+                    "Multiple hyperlinks found."
+                )
+            )
+            m.setInformativeText(
+                translate(
+                    "draft",
+                    "This may lead to the opening of various windows"
+                )
+            )
+            m.setStandardButtons(m.Ok | m.Cancel)
+            ret = m.exec_()
+
+        if len(self.hyperlinks_list) == 1 or ret == m.Ok:
+            for hyperlink in self.hyperlinks_list:
+                self.open_hyperlink(hyperlink)
+
+    def find_hyperlinks(self):
+        self.hyperlinks_list = []
+
+        for o in FreeCADGui.Selection.getCompleteSelection():
+            if hasattr(o.Object, "Text"):
+
+                for text in o.Object.Text:
+                    hyperlinks = re.findall(r"((%\w+%|/\w+|\w:\\|\w{3,4}://)[\w\\/: ]+\.[\w?$#=\\\.]+)", text)
+
+                    for hyperlink in hyperlinks:
+                        self.hyperlinks_list.append(hyperlink[0])
+            
+    def has_hyperlinks(self):
+        self.find_hyperlinks()
+        
+        return len(self.hyperlinks_list) > 0
+
+    def open_hyperlink(self, hyperlink):
+        _msg(translate("draft", "Opening hyperlink") + " " + hyperlink)
+        
+        match sys.platform:
+            case ("win32", "cygwin"):
+                os.startfile(hyperlink)
+            case "darwin":
+                subprocess.call(["open", hyperlink])
+            case "linux":
+                subprocess.call(["xdg-open", hyperlink])
```

Comment:
would this work in flatpak and snap environments?

Reply:
Just started some research about that but found nothing ATM.

Python docs doesn't say nothing about that, or maybe I simply missed it.

I'll dig deeper, worst case scenario I'll have to make and test those environments but I guess it will take me some time, never done this kind of tests before :sweat_smile:

Reply:
for flatpak I can add a wrapper for xdg-open, snap might be out of luck